packages/position/src/react/index.tsx

```diff
@@ -0,0 +1,174 @@
+>((props, ref) => {
+  return <Position position={positionFns.above} ref={ref} {...props} />
+})
+Above.displayName = 'Above'
+
+export const AboveLeft = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.aboveLeft} ref={ref} {...props} />
+})
+AboveLeft.displayName = 'AboveLeft'
+
+export const AboveRight = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.aboveRight} ref={ref} {...props} />
+})
+AboveRight.displayName = 'AboveRight'
+
+export const Below = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.below} ref={ref} {...props} />
+})
+Below.displayName = 'Below'
+
+export const BelowLeft = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.belowLeft} ref={ref} {...props} />
+})
+BelowLeft.displayName = 'BelowLeft'
+
+export const BelowRight = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.belowRight} ref={ref} {...props} />
+})
+BelowRight.displayName = 'BelowRight'
+
+export const RightOf = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.rightOf} ref={ref} {...props} />
+})
+RightOf.displayName = 'RightOf'
+
+export const LeftOf = React.forwardRef<
+  HTMLElement,
+  Omit<PositionProps, 'position'>
+>((props, ref) => {
+  return <Position position={positionFns.leftOf} ref={ref} {...props} />
+})
+LeftOf.displayName = 'LeftOf'
+
+interface ShowProps {
+  ref?: React.RefObject<HTMLElement>
+  style?: React.CSSProperties
+}
+interface PositionProps {
+  children: React.FunctionComponentElement<{}>
+  inNode?: HTMLElement
+  position: positionFns.PositionFunction
+  show: React.FunctionComponentElement<ShowProps>
+  target?: HTMLElement | React.RefObject<HTMLElement>
+  when?: boolean
+}
+const Position = React.forwardRef<HTMLElement, PositionProps>(
+  (props, forwardedRef) => {
+    const { target, position: positionFn } = props
+    const inNode = props.inNode || (canUseDOM() ? document.body : undefined)
+    const when = typeof props.when === 'boolean' ? props.when : true
+
+    const [shownOnce, setShownOnce] = React.useState(false)
+    const [style, setStyle] = React.useState<
+      Partial<positionFns.PositionStyle>
+    >({
+      position: 'absolute'
+    })
+
+    const ref = React.useRef<HTMLElement>()
+    React.useImperativeHandle(forwardedRef, () => ref.current as HTMLElement)
+
+    const child = React.Children.only(
+      props.children as React.FunctionComponentElement<{}>
+    )
+    const innerRef = React.useRef<HTMLElement>()
+    const showRef = props.show.ref || innerRef
+    const showEl = cloneElementWithRef(
```

Comment:
Used util#cloneElementWithRef instead of React.cloneElement here.  Was happier with typing.  And it seemed like we were being smarter in this util fn.